src/cascadia/TerminalControl/TermControlUiaProvider.cpp

```diff
@@ -106,25 +106,25 @@ std::deque<UiaTextRangeBase*> TermControlUiaProvider::GetSelectionRanges(_In_ IR
 
 UiaTextRangeBase* TermControlUiaProvider::CreateTextRange(_In_ IRawElementProviderSimple* const pProvider)
 {
-    return UiaTextRange::Create(_pData, pProvider);
+    return WRL::Make<UiaTextRange>(_pData, pProvider).Detach();

```

Comment:
I feel like detaching on the way out is a bad idea. I feel like the caller should detach if it wants to take responsibility for that and all of these should return something like `ComPtr<UiaTextRangeBase>` given they're helpers only referenced inside our own code.

I would only really want to detach it when giving it back on a public interface that we don't control/can't change.

Reply:
Also switching them around to return an HRESULT. That way we can pass that along to the automation client when it calls it.